src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -119,7 +120,11 @@ public DirectoryLoadContext(string directory, AnalyzerAssemblyLoader loader, Ass
 
             protected override Assembly? Load(AssemblyName assemblyName)
             {
-                var simpleName = assemblyName.Name!;
+                if (_loader.ResolveAssemblyExternally(assemblyName) is { } externallyResolvedAssembly)
```

Comment:
There are now effectively two types of "let me resolve assemblies elsewhere" patterns:

1. `IAnalyzerAssemblyResolver`
2. Just use the compiler load context

Did you consider collapsing them? Basically saying having an impl of `IAnalyzerAssemblyResolver` that looks in a specific ALC (is is done just below) then run that at the right order within these resolvers?

Reply:
Oh, that's a really interesting idea, I hadn't thought of that. Let me play with it and see what it looks like.

Reply:
@jaredpar I made the change, but ended up keeping it pretty simple.

I did go down a path where I made it the responsibility of the caller to pass in the compiler resolver. This actually simplified our loading code somewhat, because we no longer have to consider `ALC`s, but this meant that it was entirely on the host to make sure the correct resolver was passed in, and we don't really have any way of testing that. The code for that approach is here: https://github.com/chsienki/roslyn/commit/34857e5d5031db02016cf188d7c2e5d1946f3320 if you want to see what it looks like.